### DISCUSSION TITLE
theme object for light AND dark mode

### DISCUSSION THREAD

**Discussion by Zerebokep** [mantinedev/mantine#1699] [Q&A] [answered]
I currently can't wrap my head around how I should setup my theme object which includes both, colors for light and dark mode. Can anyone give me an example how it should look like?

Currently I merge the colorScheme with my theme object (theme={{ ...{ colorScheme }, ...theme }}), but I'm not sure if this is the correct approach as I can't redefine variables like primaryColor, only color.dark and color.light.

**Comment by rtivital** [comment, marked as answer]
You can create two separate objects for light and dark theme and swap them depending on color scheme:
```tsx
const darkTheme = { colorScheme: 'dark', ...otherTokens };
const lightTheme = { colorScheme: 'light', ...otherTokens };

// later...
theme={colorScheme === 'dark' ? darkTheme : lightTheme}
```
